floc.bs

```diff
@@ -1 +1,182 @@
+
+  The <dfn>string representation of the interest cohort id</dfn> is the string representation of the mapped integer of the [=interest cohort id=] in decimal (e.g. “17319”). If the [=interest cohort id=] is invalid, the string representation will be an empty string.
+
+  The <dfn>interest cohort version</dfn> identifies the <a href="#cohort-assignment-algorithm">algorithm</a> used to compute the [=interest cohort id=].
+
+  The <dfn>string representation of the interest cohort version</dfn> is [=implementation-defined=]. It’s recommended to make the browser vendor name as part of the version (e.g. “chrome.2.1”, “v21/mozilla”), so that when exposed to the Web, there won't be naming collisions across browser vendors.
+
+  The {{InterestCohort}} dictionary is used to contain the [=string representation of the interest cohort id=] and the [=string representation of the interest cohort id=].
+
+  <pre class="idl">
+  dictionary InterestCohort {
+    DOMString id;
+    DOMString version;
+  };
+  </pre>
+</section>
+
+<section>
+  <h2 id="the-api">The API</h2>
+
+  The interest cohort API lives under the {{Document}} interface since the access permission is tied to the document scope, and the API is only available if the document is in [=secure context=].
+
+  <pre class="idl">
+    partial interface Document {
+        Promise&lt;InterestCohort&gt; interestCohort();
+    };
+  </pre>
+
+  The <dfn for="Document" method>interestCohort()</dfn> method steps are:
+  1. Let |p| be [=a new promise=].
+  2. Run the following steps [=in parallel=]:
+
+    2.1. If any of the following is true:
+      - [=this=] is not [=allowed to use=] the "<code><a href="#interest-cohort-policy-controlled-feature">interest-cohort</a></code>" feature.
+      - The document is not allowed to access the interest cohort per user preference settings.
+      - The user agent believes that too many high-entropy bits of information have already been consumed by the given document, and exposing an interest cohort would violate a privacy budget.
+      - The <a href="#cohort-assignment-algorithm">cohort assignment algorithm</a> is unavailable.
+
+      then [=reject=] |p| with a "{{NotAllowedError}}" {{DOMException}} and return |p|.
+
+    2.2. Let |id| be [=interest cohort id=] from running the <a href="#cohort-assignment-algorithm">cohort assignment algorithm</a>.
+
+    2.3. Let |version| be the [=interest cohort version=] corresponding to the <a href="#cohort-assignment-algorithm">cohort assignment algorithm</a>.
+
+    2.4. Create the {{InterestCohort}} dictionary |d|, with members {{InterestCohort/id}} being the <a href="#string-representation-of-the-interest-cohort-id">string representation</a> of |id|, and {{InterestCohort/version}} being <a href="#string-representation-of-the-interest-cohort-version">string representation</a> of |version|.
+
+    2.5. [=Resolve=] |p| with |d|.
```

Comment:
You can't resolve the promise from an in-parallel section; you need to queue a task (and choose a task source for that queued task).

Reply:
Done.